### PR TITLE
Feature/dockerfile and dockerignore

### LINKED ISSUE #2: Build a container image for the Banking Web UI
As a Developer, I want to containerize the Web UI in order to make sure the app can run in a clean and homogeneous environment.

**Acceptance Criteria**
-
* Use Docker as the Container Engine.
* A Dockerfile should be created at the root of the repo.
* The Dockerfile should take advantage of the building cache.

**Deadline**
-
Monday 6th, 2023

### LINKED ISSUE #3: Enhance the building container image for Banking Web UI
As a Developer, I need to reduce the size of each container image that I built every time I add new changes into the source code.

**Acceptance Criteria**
-
* A `.dockerignore` hidden file should be created at the same level where the `Dockerfile` is.
* The `.dockerignore` file should have a list of folders and files that are not needed in the container image.

**Deadline**
-
Monday 6th, 2023

### PR DESCRIPTION
**Pull request by DMGochoa**
The Dockerfile was created and the .dockerignore. Also some changes in the main.py to specify the host.

### CODE REVIEW

**Review by aolguin89 — CHANGES_REQUESTED** [review]
Hi Diego!, it´s very good, maybe we can change the line 17 by .git* to include .gitignore file or */.git* in order to manage /log/.gitkeep empty file
Thanks great job

**Comment by DMGochoa** [comment]
Ty @aolguin89 I change the .git to .git*

File: main.py [hunk 1]
```diff
@@ -134,4 +134,4 @@ def router(url):
         return jumbotron
 
 if __name__ == "__main__":
-    app.run_server(debug=True, port=8050)
+    app.run_server(debug=True, host='0.0.0.0', port=8050)
```
Comment (calebespinoza) [inline main.py]: Disable the debug mode when running the app in the container.
Research about FLASK env vars, for instance: FLASK_DEBUG_MODE.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+COPY requirements.txt /UI_web/requirements.txt
+
+RUN pip install --no-cache-dir --upgrade -r /UI_web/requirements.txt
+
+COPY ./ /UI_web/
+# Open the port conection
+EXPOSE 8050
+# Para que el contenedor este encendido continuamente (Un while inf)
+#CMD bash -c "while true; do sleep 1; done"
```
Comment (calebespinoza) [inline Dockerfile]: A CMD instruction should be defined to run the app successlully.

File: .dockerignore [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+.coverage.*
+.cache
+nosetests.xml
+coverage.xml
+*.cover
+*.log
+.git*
+.mypy_cache
+.pytest_cache
+.hypothesis
```
Comment (calebespinoza) [inline .dockerignore:24]: Add all the files and folders that are not needed for the web ui app, like Dockerfile, .dockerignore, etc.

File: Dockerfile (hunk 2 above)
Comment (DMGochoa) [reply Dockerfile]: The CMD is now added as CMD ['python' 'main.py']

File: main.py (hunk 1 above)
Comment (DMGochoa) [reply main.py]: Now, the debug mode is off when the app runs in the container.

File: prueba.py [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import optparse
```
Comment (calebespinoza) [inline prueba.py]: Why is this file named `prueba.py` in spanish, as far as we know, we are programing in English language.

**Review by calebespinoza — APPROVED** [review]

**Review by aolguin89 — APPROVED** [review]

**Comment by calebespinoza** [comment]
Containerize web ui and implement a dockerignore file.

**Pull request merged**